File: src/front/client/src/app/Models/Forecast.ts
export class Forecast {
  date: Date;
  day: string;
  description: string;
  icon: string;
  temperature: number;
  minTemperature: number;
  maxTemperature: number;
  humidity: number;
  wind: number;

  constructor(
    date: Date,
    description: string,
    icon: string,
    temperature: number,
    minTemperature: number,
    maxTemperature: number,
    humidity: number,
    wind: number
  ) {
    this.date = date;
    this.day = this.getDayOfWeek(date);
    this.description = description;
    this.icon = icon;
    this.temperature = temperature;
    this.minTemperature = minTemperature;
    this.maxTemperature = maxTemperature;
    this.humidity = humidity;
    this.wind = wind;
  }

  private getDayOfWeek(date: Date): string {
    const days = ['Sun', 'Mon', 'Tue', 'Wed', 'Thu', 'Fri', 'Sat'];
    return days[date.getDay()];
  }
}

File: src/back/API/Services/WeatherForecast/Models/Forecast.cs
namespace API.Services.WeatherForecast.Models;

public class Forecast
{
    public DateTime Date { get; set; }
    public string? Day { get; set; }
    public string? Description { get; set; }
    public string? IconUrl { get; set; }
    public double? Temperature { get; set; }
    public double? MinTemperature { get; set; }
    public double? MaxTemperature { get; set; }
    public double? Humidity { get; set; }
    public double? WindSpeed { get; set; }
}